<launch>

<param name="use_sim_time" value="true" />


  <node
    pkg="rviz"
    type="rviz"
    name="rviz"
    args="-d $(find motion_hokuyo)/rviz_cfg/removed_floor.rviz"/>
   />
 
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
         <param name="resolution" value="0.3" />
         <param name="frame_id" type="string" value="map" />
         <param name="sensor_model/max_range" value="10.0" />
         <remap from="cloud_in" to="/pcl_test" />
    </node>

    <node pkg="motion_hokuyo" type="hokuyo_ground_filter" name="hokuyo_ground_filter"/>

    <node pkg="motion_hokuyo" type="plane_extract_ransac" name="plane_extract_ransac">
    </node>

    <node pkg="motion_hokuyo" type="max_z" name="max_z">
    	<remap from="assembled_cloud" to="/planes_pointcloud"/>
    </node>

    <node pkg="motion_hokuyo" type="motion_detector" name="motion_detector"/>

    <!--<node pkg="motion_hokuyo" type="motion_filter_v3" name="motion_filter"/>-->

    <node pkg="motion_hokuyo" type="object_detector_v3" name="object_detector"/>

    <node pkg="motion_hokuyo" type="velodyne_bound" name="velodyne_bound"/>

    <arg name="path"/> 
    <arg name="speed" default="1"/> 
    <arg name="other" default=""/> 
    <arg name="topics" default=""/> 
    <node pkg="rosbag" type="play" name="player" output="screen" args="$(arg path) -r $(arg speed) --topics /tf /velodyne_points /scan --clock" />

</launch>
